### PR TITLE
Update threshold value after new result is added

### PR DESCRIPTION
**Pull request by heemin32**
### Description
Update threshold value after new result is added

Because the threshold is not updated after inserting a new value to the result list, new value got inserted in the result regardless of its size.

### Issues Resolved
https://github.com/opensearch-project/k-NN/issues/1714
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by junqiu-lei** [comment]
Could we create a diff for the patch update? Example: https://github.com/junqiu-lei/faiss/commit/b36a28778a493df46a7f40b03cc5b426e94a4ba1

**Comment by navneet1v** [comment]
@heemin32 can you add more details on what was the root cause of the issue?

**Comment by heemin32** [comment]
> Could we create a diff for the patch update? Example: [junqiu-lei/faiss@b36a287](https://github.com/junqiu-lei/faiss/commit/b36a28778a493df46a7f40b03cc5b426e94a4ba1)

https://github.com/facebookresearch/faiss/compare/0d1385959ddecabb2825957e48ff28ff0e8abf53..da77c9972f6e7313f956111129541677b33429cb

**Review by junqiu-lei — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 * Support script score when doc value is disabled and fix misusing DISI [#1696](https://github.com/opensearch-project/k-NN/pull/1696)
 ### Bug Fixes
 * Block commas in model description [#1692](https://github.com/opensearch-project/k-NN/pull/1692)
+* Fix a bug in multi-vector with faiss engine [#1715](https://github.com/opensearch-project/k-NN/pull/1715)
```
Comment (jmazanec15) [inline CHANGELOG.md:23]: Lets come up with a better description. Also, "Fix a bug" is implied if entry is in "Bug Fixes" section

**Review by jmazanec15 — COMMENTED** [review]
@heemin32 Does this fix need to be moved to upstream as well or is it just concerned with patches?

**Comment by heemin32** [comment]
> patches

Already updated the upstream code as well.

**Review by jmazanec15 — APPROVED** [review]

**Review by junqiu-lei — APPROVED** [review]

**Pull request merged**